File: backend/src/main/java/io/metersphere/base/domain/ProjectVersion.java
package io.metersphere.base.domain;

import java.io.Serializable;
import lombok.Data;

@Data
public class ProjectVersion implements Serializable {
    private String id;

    private String projectId;

    private String name;

    private String description;

    private String status;

    private Long publishTime;

    private Long startTime;

    private Long endTime;

    private Long createTime;

    private String createUser;

    private Boolean latest;

    private static final long serialVersionUID = 1L;
}

File: frontend/src/models/projectManagement/projectVersion.ts
// 项目版本公共信息
export interface ProjectCommon {
  name: string;
  description?: string;
  status?: boolean;
  publishTime?: number;
  latest?: boolean;
  projectId: string;
}

// 项目版本项
export interface ProjectItem extends ProjectCommon {
  createTime: number;
  createUser: string;
  id: string;
}
// 项目版本选项列表
export interface ProjectVersionOption {
  name: string;
  id: string;
  latest: boolean;
  enable: boolean;
}
